<launch>
    <!--
    Convert BAG file $(arg bag_file) into several csv files, save them into $(arg csv_path)
    Additionally merge all data-sets into one file $(arg bag_file).csv for further processing
     -->
    <param name="use_sim_time" value="true" />
    <param name="input_format" value="gazebo" />  <!-- gazebo or real_data -->

    <!-- down/up sampling data: [S] sec, [L] ms, [U] us -->
    <arg name="resampling" default="50L" />

    <arg name="bag_file" />
    <arg name="output_preprocessed_file_name" default="$(arg bag_file)_preprocessed.csv" />
    <arg name="local_path" default="bags/experiments/" />
    <arg name="bag_path" default="$(find autonomous_bicycle)/$(arg local_path)" />
    <arg name="csv_path" default="$(find autonomous_bicycle)/$(arg local_path)csv/" />

    <!-- Convert ROSBAG into CSV -->
    <node pkg="autonomous_bicycle" type="rosbag2csv.py" name="rosbag2csv" output="screen">
        <param name="input_filename" type="string" value="$(arg bag_file).bag" />
        <param name="input_path" type="string" value="$(arg bag_path)" />
        <param name="output_path" type="string" value="$(arg csv_path)" />
        <param name="output_processed_filename" type="string"
          value="$(arg csv_path)$(arg bag_file)/$(arg output_preprocessed_file_name)" />
        <param name="input_format" type="string" value="$(arg input_format)" />
        <param name="resampling" type="string" value="$(arg resampling)" />
    </node>
</launch>

